evaluations/src/evaluators/regex.rs

```diff
@@ -0,0 +1,361 @@
+        (InferenceResponse::Chat(response), Datapoint::Chat(_datapoint)) => {
+            debug!("Running regex evaluation for chat response");
+            
+            // Extract text content from the response
+            let response_text = extract_text_from_chat_response(response);
+            
+            // Check if the regex pattern matches
+            let matches = regex.is_match(&response_text);
+            debug!(matches = %matches, pattern = %regex_pattern, "Chat regex evaluation completed");
+            Ok(Some(Value::Bool(matches)))
+        }
+        (InferenceResponse::Json(json_completion), Datapoint::Json(_json_inference)) => {
+            debug!("Running regex evaluation for JSON response");
+            
+            // Extract text content from the JSON response
+            let response_text = extract_text_from_json_response(json_completion);
+            
+            // Check if the regex pattern matches
+            let matches = regex.is_match(&response_text);
+            debug!(matches = %matches, pattern = %regex_pattern, "JSON regex evaluation completed");
+            Ok(Some(Value::Bool(matches)))
+        }
+        _ => {
+            let datapoint_type = match datapoint {
+                Datapoint::Chat(_) => "Chat",
+                Datapoint::Json(_) => "Json",
+            };
+            let response_type = match inference_response {
+                InferenceResponse::Chat(_) => "Chat",
+                InferenceResponse::Json(_) => "Json",
+            };
+            warn!(
+                datapoint_type = %datapoint_type,
+                response_type = %response_type,
+                "Datapoint and inference response types do not match"
+            );
+            bail!("Datapoint and inference response types do not match")
+        }
+    }
+}
+
+/// Extracts text content from a chat response
+fn extract_text_from_chat_response(response: &ChatInferenceResponse) -> String {
+    let mut text_parts = Vec::new();
+    
+    for content_block in &response.content {
+        match content_block {
+            ContentBlockChatOutput::Text(text_block) => {
+                text_parts.push(text_block.text.clone());
+            }
+            ContentBlockChatOutput::ToolCall(tool_call) => {
+                // Include tool call arguments as text
+                text_parts.push(tool_call.raw_arguments.clone());
+            }
+            ContentBlockChatOutput::Thought(thought) => {
+                // Include thought text if available
+                if let Some(text) = &thought.text {
+                    text_parts.push(text.clone());
+                }
+            }
+            ContentBlockChatOutput::Unknown { data, .. } => {
+                // Include unknown data as text if it's a string
+                if let Some(text) = data.as_str() {
+                    text_parts.push(text.to_string());
+                }
+            }
+        }
+    }
+    
+    text_parts.join(" ")
```

Comment:
@GabrielBianconi what do you think we should do here?

Reply:
Ignore every content block type except text. If the number of text blocks is not exactly one, fail. Regex against that singular block.